### PR TITLE
AMQNET-596: Added local message expiry connection uri connection property

### PR DESCRIPTION
**Pull request by cjwmorgan-sol**
Message consumer only filter expired messages when local message expiry is enabled

Added message expiration integration test

Signed-off-by: cjwmorgan-sol <Christopher.Morgan@solace.com>

### CODE REVIEW

**Comment by cjwmorgan-sol** [comment]
@michaelandrepearce and @HarvetGC please take a look.
I added some tests based off of qpid jms however I think I'll need some changes from #9 to assert the amqp frames properly.

**Comment by michaelandrepearce** [comment]
@Havret

File: src/NMS.AMQP/NmsConnectionFactory.cs [hunk 1]
```diff
@@ -99,6 +99,11 @@ private IdGenerator ConnectionIdGenerator
             }
         }
 
+        /// <summary>
+        /// Enables local message expiry for all MessageConsumers under the connection. Default is true.
+        /// </summary>
+        public bool IsLocalMessageExpiry { get; set; } = true;
```
[the remark this answers is not in the document]
Comment (michaelandrepearce) [reply src/NMS.AMQP/NmsConnectionFactory.cs]: @cjwmorgan-sol assume you've seen this?

[the remark this answers is not in the document]
Comment (cjwmorgan-sol) [reply src/NMS.AMQP/NmsConnectionFactory.cs]: I will update the properties references. I think having IsLocalMessageExpiry is closer to the meta data interface qpid jms has. However, the uri property should be the same.

**Comment by michaelandrepearce** [comment]
@cjwmorgan-sol could you squash commits and make sure jira is pre fixed on the squashed

**Comment by cjwmorgan-sol** [comment]
Squashed, hopefully.

**Comment by michaelandrepearce** [comment]
@cjwmorgan-sol great thanks, merging.

**Pull request merged**